KAN-170-모의면접

kimgorok opened the pull request:
## 😀 제목 : 모의면접 로직 추가 및 api 연결

<br/>

## 🔎 작업 내용

- 어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [x] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [x] 파일 혹은 폴더 삭제

<br/>

## 🎨 이미지 첨부

https://github.com/user-attachments/assets/5831c4b6-5004-4bec-a5ec-d79342b60837

https://github.com/user-attachments/assets/206ec4b3-dd10-490e-bccd-e7f0842f9506

<br/>

## 🔧 앞으로의 과제

- 다음으로 할 일은 무엇인가요??

해당 페이지를 url을 통해 들어오지 않도록 하는 방법을 찾아봅니다. (router.replace 사용).
상수화를 진행합니다.
api가 수정되면 이를 반영합니다.

  <br/>

## ➕ 추가 코멘트

- (10.24)
1. 비디오 부분을 다루는 컴포넌트를 Video.tsx로 분리하였습니다.

- (10.25)
1. 질문 선택 페이지 및 Progressbar를 주석 처리 하였습니다.

- (10.26)
1. 모의 면접 임시 튜토리얼을 생성하였습니다.
2. 면접기록 삭제 api를 연결하였습니다.
3. 채팅 메시지 전송 api를 연결하였습니다.
4. 과거 면접 기록 조회 페이지를 추가하였습니다.

- (10.27)
1. 질문 수정 api를 연결하였습니다.
2. 공용 store를 사용해 메시지 기록을 포스트 작성에 전달하도록 하였습니다.
3. router.replace를 사용해 자소서선택, 채팅 부분이 브라우저 히스토리에 남지 않게 하였습니다.

- (10.31)
1. 채팅 입력 창 높이 조절 훅을 userContainerHeight.ts 파일에 합쳤습니다.

- (11.02)
1. 새로운 튜토리얼 페이지를 제작하였습니다.

- (11.03)
1. 튜토리얼 페이지의 툴팁을 개선하였습니다.

- (11.05)
1. 기존의 포스트 작성 store와 채팅 메시지를 저장하는 store를 분리하였습니다.
2. 채팅 내역 포맷팅 함수를 포스트 작성 부분으로 옮겼습니다.

<br/>

## 🤔 Issue Ticket

- [Ticket](https://gwagseoyeon.atlassian.net/browse/KAN-170)

kimgorok committed:
♻️refactor: [모의면접] 비디오 컴포넌트 분리

- Video.tsx 생성 후 컴포넌트 분리

KAN-170

kimgorok committed:
🚚rename: [모의면접] 폴더명 변경

- [id]를 [usename]으로 변경
- chat 폴더 이동

KAN-170

kimgorok committed:
🚚rename: [모의면접] interviewStore를 chatStore로 변경

- 추후 ID를 관리할 store 생성을 위해 이름 변경

KAN-170

kimgorok committed:
🎉init: [모의면접] interviewStore 생성

- interviewId와 companyId를 관리할 interviewSotre.ts 생성

KAN-170

kimgorok committed:
🔥remove: [모의면접] progressbar 및 질문 선택 페이지 삭제

- ProgressBar.tsx, chatStore.ts, 질문 선택 page.tsx 주석 처리
- 자소서 선택 후 바로 채팅 페이지로 이동

KAN-170

kimgorok committed:
✨feat: [모의면접] 면접 튜토리얼 추가 및 자소서 POST 연결

- 면접 튜토리얼 컴포넌트 생성
- 면접 생성 POST를 연결(정상 작동)
- POST 후, 면접기록 쿼리 무효화

KAN-170

kimgorok committed:
🐛fix: [모의면접] 질문 생성 api 호출 시점 변경

- 페이지에 진입하면 바로 호출하던 api를, 버튼을 눌렀을 때 호출하도록 수정

KAN-170

kimgorok committed:
✨feat: [모의면접] 면접기록 삭제 api 연결

- 면접기록 삭제 api 연결
- 모달 추가

KAN-170

kimgorok committed:
✨feat: [모의면접] 채팅 메시지 전송 api 연결

- POST api 연결
- 새 질문, 꼬리 질문에도 POST 연결

KAN-170

kimgorok committed:
🎉init: [모의면접] 면접 기록 조회 페이지 추가

- 면접 기록 페이지 추가 및 api 연결
- api 엔드포인트 수정
- store에 Id 저장 후, 기록 조회 페이지에서 사용

KAN-170

kimgorok committed:
✨feat: [모의면접] 질문 수정 api 연결

- 질문 수정 api 연결
- 키값이 없던 map 함수에 uuid 키값 연결

KAN-170

kimgorok committed:
🐛fix: [모의면접] 기업 목록 공백도 검색 가능하게 설정

- 공백을 검색할 시 모든 기업 목록 출력

KAN-170

kimgorok committed:
✨feat: [모의면접] 포스트 작성에 채팅 내역 전달

- store를 사용해 채팅 내역 저장 후, 포스트 작성 시 기본 값으로 지정

KAN-170

kimgorok committed:
♻refactor: [모의면접] 새 질문, 꼬리 질문 버튼 컴포넌트 분리

- TailNewBtn.tsx 컴포넌트 생성 및 적용

KAN-170

kimgorok committed:
🐛fix: [모의면접] 면접 부분이 브라우저 히스토리에 안남게 수정

- router.push등을 replace로 수정

KAN-170

BaekJiyeon02 commented on src/app/_store/postWirte.ts:
```diff
@@ -0,0 +1,32 @@
+    const formattedMessages = messages.map((msg) => `${msg.isAI ? '면접관' : '나'}: ${msg.text}`).join('\n\n');
+    set({ postData: formattedMessages });
+  },
+}));
```
채팅 내용을 글쓰기로 잘 옮기셨네요!

BaekJiyeon02 reviewed (approved):
고생많으셨습니다!

kimgorok committed:
💡comment: [모의면접] 포스트 작성 store 주석 추가

- postWrite.ts에 주석 추가

KAN-170

kimgorok committed:
💡comment: [모의면접] ProgressBar 주석 처리

- ProgressBar.tsx 전문 주석 처리

KAN-170

kimgorok committed:
🐛fix: globals.css 수정

- 잘못 선언된 부분 수정

KAN-170

kimgorok commented on src/app/globals.css:
```diff
@@ -75,7 +75,6 @@ body {
   /* 블로그 */
   .blog-favor-frame {
-    @apply flex items-center gap-[5px] px-3 py-1.5 cursor-pointer;
     @apply flex items-center gap-[5px] px-3 py-[5px] cursor-pointer;
```
계속 충돌이 생기는 부분이었습니다

kimgorok committed:
♻️refactor: [모의면접] 채팅창 높이조절 훅 분리

- 채팅창 높이 조절 ref도 useContainerHeight.ts에 합침

KAN-170

kimgorok committed:
🔥remove: [모의면접] 사용하지 않는 파일 제거

- ProgressBar.tsx, QuestionBox.tsx, chatStore.ts, page.tsx

KAN-170

kimgorok commented:
(11.01)
- 튜토리얼 수정사항을 반영한 뒤, 다시 커밋하겠습니다

kimgorok commented on src/app/(route)/(interview)/interview/[username]/[corp]/_components/TutorialBox.tsx:
```diff
@@ -0,0 +1,78 @@
+import { useRef, useState } from 'react';
```
(11.01)
- 튜토리얼 수정사항을 반영한 뒤, 다시 커밋하겠습니다

kimgorok committed:
🎉init: [모의면접] 튜토리얼 페이지 레이아웃 세팅

- 튜토리얼 페이지 레이아웃 세팅

KAN-170

kimgorok committed:
✨feat: [모의면접] 튜토리얼 페이지 레이아웃 제작

- assets 이미지 추가
- ToolTip.tsx 컴포넌트 작성
- 재생, 새로고침 아이콘 추가
- 튜토리얼 페이지 state 적용

KAN-170

kimgorok committed:
✨feat: [모의면접] 튜토리얼 주소 이동 추가

- 자소서 페이지에서 튜토리얼 페이지로 이동
- 튜토리얼 페이지에서 채팅 페이지로 이동

KAN-170

kimgorok committed:
🔥remove: [모의면접] 기존의 튜토리얼 컴포넌트 삭제

- TutorialBox.tsx 삭제
- 자소서 페이지에 적용

KAN-170

kimgorok committed:
💄design: [모의면접] 튜토리얼 툴팁 UI 개선

- 툴팁의 등장 애니메이션 추가
- 툴팁의 위치 변경
- messageVariants.ts 파일명 변경 및 폴더 위치 변경

KAN-170

kimgorok committed:
💄design: [모의면접] globals.css 일부 정리

- tail-new-facial-icon, interview-history, interview-history-text 추가

KAN-170

kimgorok committed:
🎨style: [모의면접] 튜토리얼 코드 포맷팅

- 기타 코드 포맷팅

KAN-170

kimgorok commented on src/app/(route)/(interview)/interview/[username]/[corp]/tutorial/page.tsx:
```diff
@@ -0,0 +1,201 @@
+'use client';
```
기존의 튜토리얼 대신, 새로 추가된 튜토리얼 페이지 입니다.

kimgorok committed:
✨feat: [모의면접] 채팅 내역 store 분리

- postWrite.ts의 메시지 내용 저장 부분을 chatWrite.ts로 분리

KAN-170

kimgorok committed:
🐛fix: [모의면접] 채팅 store 포맷팅 위치 수정

- useChatWriteStore 단순화
- PostWrite page에서 채팅 메시지 포맷팅

KAN-170
